platformics/api/core/strawberry_extensions.py

```diff
@@ -35,6 +36,52 @@ def get_func_with_only_deps(func: typing.Callable[..., typing.Any]) -> typing.Ca
     return newfunc
 
 
+class RegisteredPlatformicsPlugins:
+    plugins: dict[str, typing.Callable[..., typing.Any]] = {}
+
+    @classmethod
+    def register(cls, callback_order: str, type: str, action: str, callback: typing.Callable[..., typing.Any]) -> None:
+        cls.plugins[f"{callback_order}:{type}:{action}"] = callback
+
+    @classmethod
+    def getCallback(cls, callback_order: str, type: str, action: str) -> typing.Callable[..., typing.Any] | None:
+        return cls.plugins.get(f"{callback_order}:{type}:{action}")
+
+
+def register_plugin(callback_order: str, type: str, action: str) -> Callable[..., Callable[..., Any]]:
+    def decorator_register(func: Callable[..., Any]) -> Callable[..., Any]:
+        RegisteredPlatformicsPlugins.register(callback_order, type, action, func)
+        return func
+
+    return decorator_register
+
+
+class PlatformicsPluginExtension(FieldExtension):
+    def __init__(self, type: str, action: str) -> None:
+        self.type = type
+        self.action = action
+        self.strawberry_field_names = ["self"]
+
+    async def resolve_async(
+        self,
+        next_: typing.Callable[..., typing.Any],
+        source: typing.Any,
+        info: Info,
+        **kwargs: dict[str, typing.Any],
+    ) -> typing.Any:
+        before_callback = RegisteredPlatformicsPlugins.getCallback("before", self.type, self.action)
+        if before_callback:
+            before_callback(source, info, **kwargs)
+
+        result = await next_(source, info, **kwargs)
+
+        after_callback = RegisteredPlatformicsPlugins.getCallback("after", self.type, self.action)
+        if after_callback:
+            result = after_callback(result, source, info, **kwargs)
```

Comment:
should before and after callbacks accept the same number of args?

Reply:
I think we'd also send the `result` to the after callbacks so that they can manipulate the result